### PR TITLE
samples: drivers: Convert to new DT_INST macros

### PR DESCRIPTION
**Pull request by galak**
Convert older DT_INST_ macro use the new include/devicetree.h
DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: samples/drivers/spi_flash/src/main.c [hunk 1]
```diff
@@ -14,11 +14,11 @@
 /* NB: W25Q16DV is a JEDEC spi-nor device, but has a separate driver. */
 #define FLASH_DEVICE CONFIG_SPI_FLASH_W25QXXDV_DRV_NAME
 #define FLASH_NAME "W25QXXDV"
-#elif (CONFIG_SPI_NOR - 0) || defined(DT_INST_0_JEDEC_SPI_NOR_LABEL)
-#define FLASH_DEVICE DT_INST_0_JEDEC_SPI_NOR_LABEL
+#elif (CONFIG_SPI_NOR - 0) || DT_NODE_HAS_PROP(DT_INST(0, jedec_spi_nor), label)
```
Comment (henrikbrixandersen) [inline samples/drivers/spi_flash/src/main.c]: Better handled as node existence checks?

Comment (pabigot) [reply samples/drivers/spi_flash/src/main.c]: I agree but I believe these are automated conversions.  We can probably do a replacement as a separate cleanup after everything's completed.  I suspect there are more like this, or will be soon.

**Review by pabigot — APPROVED** [review]

File: samples/drivers/spi_flash/src/main.c (hunk 1 above)
Comment (galak) [reply samples/drivers/spi_flash/src/main.c]: I fixed this up,  There are a small handful of cases in which we test for label that I'll send some cleanup PRs for.

**Review by vanwinkeljan — APPROVED** [review]
+1 on the display changes

**Comment by galak** [comment]
@vanwinkeljan I changed the display sample to remove dts_fixup.h and merge that into the .c file.  let me know if you've go any issue with that.

**Comment by vanwinkeljan** [comment]
@galak Fine for me now, it is only a stopgap solution until we have a mechanism to detect a display in the system

**Pull request merged**